### PR TITLE
Added samples of nested formats

### PR DESCRIPTION
**Pull request by madphysicist**
Setting precision, width, etc. using positional and keyword arguments instead of fixing them in the string literal being formatted. This is something that I have always been uncertain about, so I made this PR.

### CODE REVIEW

File: tests/test_content.py [hunk 1]
```diff
@@ -421,6 +421,83 @@ def test_datetime():
     assert new_result == '2001-02-03 04:05'  # output
 
 
+def test_nested_align():
+    """
+    # Nested Formats
```
Comment (zerok) [inline tests/test_content.py]: Hi :) Thank you very much for the patch! Could you just please replace this heading with "Nested Formats" as all other headings so far are using sentence-formatting? Thanks!

Comment (ulope) [inline tests/test_content.py]: I would even suggest not calling it "nested formats" but rather "Parametrized Formats" or something similar.

File: tests/test_content.py [hunk 2]
```diff
@@ -421,6 +421,83 @@ def test_datetime():
...
+    the format to be specified dynamically using nested formats.
+
+    Nested formats can be applied to every portion of the format after
+    the colon. Variable names can not be set dynamically because double
+    braces are the escape characters for braces, so something like
+    `'{{which}}'.format(0.1, 0.2, which=1)` would result in `'{which}'`,
+    not in `'0.2'`. Braces are not allowed in attribute names and item
+    indices are passed literally, so nested formats can not be used
+    anywhere before the colon in a format.
+
```
Comment (ulope) [inline tests/test_content.py]: I feel this paragraph doesn't really help the explanation. Maybe just the first sentence would be enough.

**Comment by ulope** [comment]
Hi. Yes, thanks for helping out :)

I agree that we need to improve the explanation on this aspect of formatting. We do have it sprinkled in a tiny bit with some ([1](https://github.com/ulope/pyformat.info/blob/master/tests/test_content.py#L124), [2](https://github.com/ulope/pyformat.info/blob/master/tests/test_content.py#L180)) of the other examples but no clear explanation how it works. I think we can remove those once this is merged since they will be pretty much duplicates at this point.

I've added some comments inline, but otherwise think this looks great.

File: tests/test_content.py (hunk 1 above)
Comment (madphysicist) [inline tests/test_content.py]: @zerok. I am not sure that I understand the request. Do you want me to remove the hash? It seems consistent with all the other headings.

**Comment by madphysicist** [comment]
I have addressed all the comments except @zerok's initial request. Let me know if you want me to make further changes.

File: tests/test_content.py [hunk 3]
```diff
@@ -421,6 +395,77 @@ def test_datetime():
...
 
 
+def test_param_align():
+    """
+    # Parametrized Formats
+
+    Additionally, new style formatting allows all of the components of
+    the format to be specified dynamically using parametrization. Parametrized
+    formats are nested expressions in braces that can appear anywhere in the
+    parent the format after the colon.
```
Comment (ulope) [inline tests/test_content.py]: "the parent the format" doesn't look right :)

**Comment by ulope** [comment]
LGTM (apart from the small thing I commented on above).

@zerok What do you think?

**Comment by madphysicist** [comment]
Addressed your comment.

File: tests/test_content.py [hunk 4]
```diff
@@ -421,6 +395,77 @@ def test_datetime():
     assert new_result == '2001-02-03 04:05'  # output
 
 
+def test_param_align():
+    """
+    # Parametrized Formats
```
Comment (zerok) [inline tests/test_content.py]: Please write "Parametrized formats" instead of "Parametrized Formats" here to be more consistent with other headings 😃

**Comment by zerok** [comment]
@madphysicist @ulope Sorry for the late response. I've just added one more comment. Other than that this looks great! Thank you very much!

File: tests/test_content.py (hunk 4 above)
Comment (madphysicist) [inline tests/test_content.py]: Done. Now I understand what you meant by sentence formatting. Your original comment had the F capitalized.

**Comment by ulope** [comment]
Ok this looks great. Again, thank you very much for the contribution.

**Comment by madphysicist** [comment]
You're very welcome. You made a great reference that I use often. I'm glad to be a part of it.

**Comment by zerok** [comment]
@madphysicist arghhh sorry :)

**Pull request merged**